src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt

```diff
@@ -147,23 +148,31 @@ internal class RiskService(
   }
 
   private fun createTimeLineDataPoint(riskScoreResponse: RiskScoreResponse?): PredictorScore {
-    val rsr = riskScoreResponse?.riskOfSeriousRecidivismScore
-    val osp = riskScoreResponse?.sexualPredictorScore
-    val ogp = riskScoreResponse?.generalPredictorScore
-    val ogrs = riskScoreResponse?.groupReconvictionScore
-    val ovp = riskScoreResponse?.violencePredictorScore
+    val rsrScore = riskScoreResponse?.riskOfSeriousRecidivismScore
+    val rsr = if (rsrScore?.scoreLevel == null && rsrScore?.percentageScore == null) null else riskScoreResponse.riskOfSeriousRecidivismScore
+    val ospScore = riskScoreResponse?.sexualPredictorScore
+    val noOspcScore = (ospScore?.ospContactScoreLevel == null && ospScore?.ospContactPercentageScore == null) || ospScore.ospContactScoreLevel.equals(OSPC_SCORE_NOT_APPLICABLE, ignoreCase = true)
+    val ospc = if (noOspcScore) null else ospScore
+    val noOspiScore = (ospScore?.ospIndecentScoreLevel == null && ospScore?.ospIndecentPercentageScore == null) || ospScore.ospIndecentScoreLevel.equals(OSPC_SCORE_NOT_APPLICABLE, ignoreCase = true)
+    val ospi = if (noOspiScore) null else ospScore
+    val ogpScore = riskScoreResponse?.generalPredictorScore
+    val ogp = if (ogpScore?.ogp1Year == null && ogpScore?.ogp2Year == null && ogpScore?.ogpRisk == null) null else ogpScore
```

Comment:
I think this property name got standardised to `twoYears`

Reply:
This represents what ARN sends us, so this is correct. The response we send to the front end will have the correct names.